### PR TITLE
Create new module to support the Rust lambda[CDS-761]

### PR DESCRIPTION
**Pull request by guyrenny**
# Description
Create a new module to support the new Rust lambda, the module will support:

S3 --> cloudtrail, vpcflow, s3-sns, cloudtrail-sns, S3Csv
Cloudwatch

 Fixes #

# How Has This Been Tested?

# Checklist:
- [ ] I have updated the relevant example in the examples directory for the module.
- [ ] I have updated the relevant test file for the module.
- [ ] This change does not affect module (e.g. it's readme file change)

### CODE REVIEW

File: examples/coralogix-aws-shipper/variables.tf
Comment (juan-coralogix) [inline examples/coralogix-aws-shipper/variables.tf:1]: I dont see the integration for the sns, not sure if is intended.
Also lots of file are missing the last empty line

File: modules/coralogix-aws-shipper/README.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+
+### Integration S3/CloudTrail/VpcFlow/S3Csv configuration
+
+| Name | Description | Type | Default | Required |
+|------|-------------|------|---------|:--------:|
+| <a name="input_s3_bucket_name"></a> [s3\_bucket\_name](#input\_s3\_bucket\_name) | The name of the S3 bucket to watch | `string` | n/a | yes |
+| <a name="input_s3_key_prefix"></a> [s3\_key\_prefix](#input\_s3\_key\_prefix) | The S3 path prefix to watch | `string` |  n/a | no |
+| <a name="input_s3_key_suffix"></a> [s3\_key\_suffix](#input\_s3\_key\_suffix) | The S3 path suffix to watch | `string` |  n/a` | no |
+| <a name="input_sns_topic_name"></a> [sns_topic_name](#input\_sns\_topic\_name) | The SNS topic that will contain the SNS subscription, need only if you use the sns integration | `string` |  n/a | no |
```
Comment (juan-coralogix) [inline modules/coralogix-aws-shipper/README.md:47]: I think we need to move this to another part as now we have sns-s3 and sns-text

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
+# Contributing Guide
```
Comment (royfur) [inline CONTRIBUTING.md:1]: This file need to be written from scratch for TF

File: modules/coralogix-aws-shipper/main.tf [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+  coralogix_regions = {
+    Europe    = "coralogix.com"
+    Europe2   = "eu2.coralogix.com"
+    India     = "coralogix.in"
+    Singapore = "coralogixsg.com"
+    US        = "coralogix.us"
+    US2       = "cx498.coralogix.com"
+  }
+
+  coralogix_url_seffix = "/api/v1/logs"
```
Comment (royfur) [inline modules/coralogix-aws-shipper/main.tf:14]: this is not needed

File: modules/coralogix-aws-shipper/main.tf [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+#### s3  integration resources ####
+###################################
+
+resource "aws_s3_bucket_notification" "lambda_notification" {
+  count  = var.integration_type == "CloudWatch" ? 0 : local.sns_enable == false ? 1 : 0
+  bucket = data.aws_s3_bucket.this[0].bucket
+  lambda_function {
+    lambda_function_arn = module.lambda.lambda_function_arn
+    events              = ["s3:ObjectCreated:*"]
+    filter_prefix       = var.integration_type == "S3" || var.s3_key_prefix != null ? var.s3_key_prefix : "AWSLogs/"
```
Comment (royfur) [inline modules/coralogix-aws-shipper/main.tf:184]: it should be set to AWSLogs/ only for CloudTrail

File: modules/coralogix-aws-shipper/main.tf [hunk 5]
```diff
@@ -0,0 +1,260 @@
...
+  coralogix_url_seffix = "/api/v1/logs"
+
+  tags = {
+    Provider = "Coralogix"
+    License  = "Apache-2.0"
+  }
+
+  s3_suffix_map = {
+    CloudTrail    = ".json.gz"
+    VpcFlow = ".log.gz"
```
Comment (royfur) [inline modules/coralogix-aws-shipper/main.tf:23]: it should be also .json.gz

File: modules/coralogix-aws-shipper/main.tf [hunk 6]
```diff
@@ -0,0 +1,260 @@
+locals {
+
+  function_name = "Coralogix-${var.integration_type}-${random_string.this.result}"
+
+  coralogix_regions = {
```
[the remark this answers is not in the document]
Comment (guyrenny) [reply modules/coralogix-aws-shipper/main.tf:5]: If you are referring to the coralogix_domains in the locals module, then if this integration is going to replace the s3 and cloudwatch we won't need this module so and we could delete him, so i wanted this integration to be ready for this.

File: modules/coralogix-aws-shipper/main.tf (hunk 5 above)
Comment (royfur) [reply modules/coralogix-aws-shipper/main.tf:23]: please check it from old lambda it looks like it is .json.gz might be mistake in the old one as well

**Review by royfur — APPROVED** [review]

**Pull request merged**